keras/engine/training.py

```diff
@@ -987,7 +987,8 @@ def fit(self,
             if the model has named inputs.
           - A `tf.data` dataset. Should return a tuple
             of either `(inputs, targets)` or
-            `(inputs, targets, sample_weights)`.
+            `(inputs, targets, sample_weights)`.  Note: datasets returning
```

Comment:
This is true for built-in models but it's not accurate for any custom model. In a custom model, you decide what's accepted: the `train_step(data)` method that you override take an entire batch from the datatset as input.